core/iwasm/common/wasm_application.c

```diff
@@ -107,7 +107,12 @@ execute_main(WASMModuleInstanceCommon *module_inst, int32 argc, char *argv[])
        the actual main function. Directly calling main function
        may cause exception thrown. */
     if ((func = wasm_runtime_lookup_wasi_start_function(module_inst))) {
-        return wasm_runtime_call_wasm(exec_env, func, 0, NULL);
+        int ret = wasm_runtime_call_wasm(exec_env, func, 0, NULL);
+#if WASM_ENABLE_THREAD_MGR != 0
+        if (!ret && wasm_runtime_get_exception(module_inst))
+            wasm_cluster_spread_exception(exec_env);
```

Comment:
Should we apply the same changes to the main branch?
And should we ignore spreading the exception "wasi proc exit" since it is a normal behavior and throwing this exception is a special handling of WAMR?

Reply:
This change is to propagate `proc_exit` and traps from the main thread, I can put an additional `WASM_ENABLE_LIB_WASI_THREADS != 0` at L111 to avoid changing the normal behavior when wasi threads are not involved

Reply:
How about applying changes in wasm_runtime_call_wasm since here it is to call the _start function of wasm app, but developer may call other functions to start the app which may also create new threads. 

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/common/wasm_runtime_common.c#L1788-L1789

For example, before return false, add:
```C
#if WASM_ENABLE_THREAD_MGR != 0 && WASM_ENABLE_LIB_WASI_THREADS != 0
        if (wasm_runtime_get_exception(module_inst))
            wasm_cluster_spread_exception(exec_env);
#endif
```

Reply:
yes that makes more sense, thanks